internal/engine/wazevo/backend/isa/arm64/machine_relocation.go

```diff
@@ -11,14 +11,22 @@ import (
 //
 // TODO: unit test!
 func (m *machine) ResolveRelocations(refToBinaryOffset map[ssa.FuncRef]int, binary []byte, relocations []backend.RelocationInfo) {
+	base := uintptr(unsafe.Pointer(&binary[0]))
 	for _, r := range relocations {
 		instrOffset := r.Offset
 		calleeFnOffset := refToBinaryOffset[r.FuncRef]
 		brInstr := binary[instrOffset : instrOffset+4]
 		diff := int64(calleeFnOffset) - (instrOffset)
 		// Check if the diff is within the range of the branch instruction.
 		if diff < -(1<<25)*4 || diff > ((1<<25)-1)*4 {
-			panic(fmt.Sprintf("TODO: too large binary where branch target is out of the supported range +/-128MB: %#x", diff))
+			// If the diff is out of range, we need to use a trampoline.
+			diff = int64(r.TrampolineOffset) - instrOffset
+			// The trampoline invokes the function using the BLR instruction
+			// using the absolute address of the callee function.
+			absoluteCalleeFnAddress := uint(base) + uint(calleeFnOffset)
+			// The trampoline should return to the next instruction after the branch instruction.
+			returnOffset := -diff + 8
```

Comment:
I don't think this is necessary by using `BR`, and not `BLR` in the trampoline

Reply:
clever!